vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identityprovider/client/DefaultIdentityDocumentClient.java

```diff
@@ -0,0 +1,130 @@
+import org.apache.http.impl.client.DefaultHttpRequestRetryHandler;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.eclipse.jetty.http.HttpStatus;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.util.function.Supplier;
+
+/**
+ * Default implementation of {@link IdentityDocumentClient}
+ *
+ * @author bjorncs
+ */
+public class DefaultIdentityDocumentClient implements IdentityDocumentClient {
+
+    private static final String IDENTITY_DOCUMENT_API = "/athenz/v1/provider/identity-document/";
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private final Supplier<SSLContext> sslContextSupplier;
+    private final HostnameVerifier hostnameVerifier;
+    private final URI configserverUri;
+
+    public DefaultIdentityDocumentClient(URI configserverUri,
+                                         SSLContext sslContext,
+                                         HostnameVerifier hostnameVerifier) {
+        this.configserverUri = configserverUri;
+        this.sslContextSupplier = () -> sslContext;
+        this.hostnameVerifier = hostnameVerifier;
+    }
+
+    public DefaultIdentityDocumentClient(URI configserverUri,
+                                         ServiceIdentityProvider identityProvider,
+                                         HostnameVerifier hostnameVerifier) {
+        this.configserverUri = configserverUri;
+        this.sslContextSupplier = identityProvider::getIdentitySslContext;
+        this.hostnameVerifier = hostnameVerifier;
+    }
+
+    @Override
+    public SignedIdentityDocument getNodeIdentityDocument(String host) {
+        return getIdentityDocument(host, "node");
+    }
+
+    @Override
+    public SignedIdentityDocument getTenantIdentityDocument(String host) {
+        return getIdentityDocument(host, "tenant");
+    }
+
+    private SignedIdentityDocument getIdentityDocument(String host, String type) {
+
+        try (CloseableHttpClient client = createHttpClient(sslContextSupplier.get(), hostnameVerifier)) {
+            URI uri = configserverUri
+                    .resolve(IDENTITY_DOCUMENT_API)
+                    .resolve(type + '/')
+                    .resolve(host);
+            HttpUriRequest request = RequestBuilder.get()
+                    .setUri(uri)
+                    .addHeader("Connection", "close")
+                    .addHeader("Accept", "application/json")
+                    .build();
+            try (CloseableHttpResponse response = client.execute(request)) {
+                String responseContent = EntityUtils.toString(response.getEntity());
+                if (HttpStatus.isSuccess(response.getStatusLine().getStatusCode())) {
+                    com.yahoo.vespa.athenz.identityprovider.api.bindings.SignedIdentityDocument entity =
```

Comment:
Is the "document" part of the name necessary elsewhere (SignedIdentityDocument -> SignedIdentity)? Sounds like "document" is something one would call a serialized and portable ASCII representation of the identity, which would be perfect for the entity/response/resource, but not internally when deserialized and represented by a proper Java object.

Reply:
_Signed identity document_, _identity document_ and _instance_ are all concepts from Athenz. The concept _identity document_ is used multiple places in the Athenz identity provider spec and our implementation.